File: fixtures/scale/csharp/large/Families/Substitution/SubstitutionDescriptor.cs
using System.Collections.Generic;
using CipherLab.Abstractions;

namespace CipherLab.Families.Substitution
{
    public class SubstitutionDescriptor : IFamilyDescriptor
    {
        public string Family => "substitution";
        public string Suite => "classical";
        public CipherKind Kind => CipherKind.Letter;

        public ICipher Cipher()
        {
            return new SubstitutionCipher();
        }

        public List<TestVector> Vectors()
        {
            return SubstitutionVectors.All();
        }
    }
}

File: fixtures/scale/swift/large/Families/Substitution/SubstitutionDescriptor.swift
struct SubstitutionDescriptor: FamilyDescriptor {
    var family: String { "substitution" }
    var suite: String { "classical" }
    var kind: CipherKind { .letter }

    func cipher() -> Cipher {
        return SubstitutionCipher()
    }

    func vectors() -> [TestVector] {
        return SubstitutionVectors.all()
    }
}
